kernel/fdtable.c

```diff
@@ -461,3 +463,66 @@ myst_fdtable_t* myst_fdtable_current(void)
     myst_assume(thread->fdtable);
     return thread->fdtable;
 }
+
+static const char* _type_name(myst_fdtable_type_t type)
+{
+    switch (type)
+    {
+        case MYST_FDTABLE_TYPE_TTY:
```

Comment:
This function can be replaced with a simple array lookup if we define a string array:
```c
char* fd_type_names = {"none", "tty", "file", "pipe", ...];
...
char* fd_type_name = fd_type_names[type];
```

Reply:
I opted for this approach so that we can get a compiler error of someone adds a new type but forgets to update this function. For example:

```
fdtable.c:469:5: error: enumeration value ‘MYST_FDTABLE_TYPE_MYNEW_TYPE’ not handled in switch [-Werror=switch]
```

Reply:
The common practice is to declare `fd_type_names` alongside `enum myst_fdtable_type`  and define `MYST_FDTABLE_TYPE_MAX` to be the last member of the enum, and use a static assertion to make sure both lists have the same size.

Reply:
It still does not allow the compiler to detect when one adds a new tag to the enum. Also, adding MYST_FDTABLE_TYPE_MAX allows that value to be passed wherever there is a type parameter which now requires checking that the type != MYST_FDTABLE_TYPE_MAX.

Reply:
This is exactly how we coded it to work with oe_result_t.